File: Part1_강의자료/Ch03_배열/문제별코드/3273_두수의합/3273.cpp
#include <stdio.h>
typedef long long ll;

int N, X;
int a[100000];
int cnt[1000001];

int main()
{
    ll ans = 0;

    scanf("%d", &N);
    for (int i = 0; i < N; i++)
        scanf("%d", a + i);
    scanf("%d", &X);

    for (int i = 0; i < N; i++)
        cnt[a[i]]++;

    for (int i = 1; i <= (X - 1) / 2; i++)
        if (X - i <= 1000000)
            ans += (ll)cnt[i] * cnt[X - i];

    printf("%lld\n", ans);
    return 0;
}

File: Part1_강의자료/Ch03_배열/문제별코드/3273_두수의합/3273.py
N = int(input())
a = list(map(int, input().split()))
X = int(input())
cnt = [0] * 1000001
ans = 0

for i in range(N):
    cnt[a[i]] += 1

for i in range(1, (X - 1) // 2 + 1):
    if X - i <= 1000000:
        ans += cnt[i] * cnt[X - i]

print(ans)
